### PR TITLE
docs: updated docstring of get_data method in base_transfromer.py

### PR DESCRIPTION
**Pull request by ayub02**
**Description:** Updated the documentation of ``get_data`` method.

### CODE REVIEW

**Comment by edx-webhook** [comment]
Thanks for the pull request, @ayub02! I've created [OSPR-5873](https://openedx.atlassian.net/browse/OSPR-5873) to keep track of it in JIRA, where we prioritize reviews. Please note that it may take us up to several weeks or months to complete a review and merge your PR.

Feel free to add as much of the following information to the ticket:
- supporting documentation
- [Open edX discussion forum threads](https://discuss.openedx.org/)
- timeline information ("this must be merged by XX date", and why that is)
- partner information ("this is a course on edx.org")
- any other information that can help Product understand the context for the PR

All technical communication about the code itself will be done via the GitHub pull request interface. As a reminder, [our process documentation is here](http://edx-developer-guide.readthedocs.org/en/latest/process/overview.html).


Please let us know once your PR is ready for our review and all tests are green.

File: event_routing_backends/processors/mixins/base_transformer.py [hunk 1]
```diff
@@ -130,27 +130,28 @@ def extract_username(self):
...
+        empty, the function will raise an exception. If required is False and key is not found or is empty, the
+        function will log the instance.
 
-            {
-                'key_a': {
-                    'key_b': {
-                        'key_c': 'final value'
+        Parameters:
+                key (str): dotted-separated key string
+                required (bool) : whether or not key is a required field (default: None)
```
Comment (ziafazal) [inline event_routing_backends/processors/mixins/base_transformer.py:141]: default is `False` instead of `None`

**Review by ziafazal — COMMENTED** [review]
@ayub02 one change needed here

**Comment by natabene** [comment]
@ayub02 Thank you for your contribution. @ziafazal I assume you will review and merge this.

**Review by ziafazal — APPROVED** [review]
👍

**Comment by edx-webhook** [comment]
@ayub02 🎉 Your pull request was merged!

Please take a moment to [answer a two question survey](https://docs.google.com/forms/d/e/1FAIpQLSceJOyGJ6JOzfy6lyR3T7EW_71OWUnNQXp68Fymsk3MkNoSDg/viewform?usp=pp_url&entry.1671973413=edx/event-routing-backends&entry.867055334=https://github.com/edx/event-routing-backends/pull/86&entry.1484655318=https://github.com/ayub02&entry.752974735=2021-06-21+11:15&entry.1917517419=2021-06-22+10:13&entry.2133058324=Yes) so we can improve your experience in the future.

**Pull request merged**